tests/lib/__init__.py

```diff
@@ -1052,8 +1052,16 @@ def hello():
         path.parent.mkdir(exist_ok=True, parents=True)
         path.write_bytes(ensure_binary(files[fname]))
 
+    # The base_dir cast is required to make `shutil.make_archive()` use
+    # Unicode paths on Python 2, making it able to properly archive
+    # files with non-ASCII names.
     retval = script.scratch_path / archive_name
-    generated = shutil.make_archive(retval, 'zip', script.temp_path)
+    generated = shutil.make_archive(
+        retval,
+        'zip',
+        root_dir=script.temp_path,
+        base_dir=text_type(os.curdir),
```

Comment:
Oh, that is also a tricky one :o

Reply:
It was so much "fun" to try to fix this. 😅

That was definitely the word we were using to describe the output.